src/app/auth/services/bulk-upload.service.ts

```diff
@@ -44,15 +48,23 @@ export class BulkUploadService {
     }
     return this.http.get(Url);
   }
-  downloadFile(): Promise<boolean> {
+  downloadFile(type: string): Promise<boolean> {
     return new Promise((resolve, reject) => {
       try {
         const anchor = document.createElement('a');
-        anchor.href = this.filePath;
-        anchor.download = this.downloadedFileName;
-        document.body.appendChild(anchor);
-        anchor.click();
-        document.body.removeChild(anchor);
+        if (type === BulkUploadType.Reads) {
```

Comment:
Here, the implementation is duplicated. It should now be something like 

```suggestion
     const filename = this.templates[type]
     if(!filename) throw new Error('Invalid template');
```